javascripts/content_scripts/posting_button.js

```diff
@@ -1,131 +1,414 @@
 /*global privlyUrlReceiptNode:false, pendingPost:false, chrome:false, ls:true,  */
 
+// This content-script provides Privly Posting Button feature.
+// It listens events of editable elements (textarea / contentEditable elements)
+// and create or show the posting button.
+
+(function() {
+
+var BUTTON_WIDTH = 16;
+var BUTTON_HEIGHT = 16;
+var BUTTON_MARGIN = 5;
+var INTERVAL_UPDATE_POSITION = 500;
+var HIDE_DELAY = 5000; // fade out after 5000 ms
+var HIDE_DELAY_SHORT = 500; // fade out after 500 ms when lose focus
+
+// create a unique className for privly button
+// will be used to easily find the button from target container
+var privlyButtonClassName = "privlyButton-" + Math.floor(Math.random() * 0xFFFFFFFF).toString(16);
+
 /**
- * Shows a button in editable content areas for starting a Privly Post.
+ * Helper methods to calculate correct position of an element.
+ * Mainly ported from jQuery.fn.position()
  */
-function addPrivlyButton() {
+var PositionHelper = {
+  /**
+   * Get computed style of an element
+   * 
+   * @param  {Element} elem
+   * @param  {string}  property
+   * @param  {boolean} numeric  Should value return as a numeric value?
+   * @return {string|Number}
+   */
+  css: function(elem, property, numeric) {
+    var value = getComputedStyle(elem)[property];
+    if (numeric) {
+      if (value === "") {
+        return 0;
+      } else {
+        return parseFloat(value);
+      }
+    } else {
+      return value;
+    }
+  },
 
-  // Create a div to handle the privly button
-  var div = document.createElement("div");
+  /**
+   * Get the closest ancestor element that is positioned
+   * 
+   * jQuery.fn.offsetParent()
+   * Ported from: https://github.com/jquery/jquery/blob/master/src/offset.js
+   * 
+   * @param  {Element} elem
+   * @return {Element}
+   */
+  offsetParent: function(elem) {
+    var offsetParent = elem.offsetParent || document.documentElement;
+    while (offsetParent && (offsetParent.nodeName !== "HTML" && PositionHelper.css(offsetParent, "position") === "static")) {
+      offsetParent = offsetParent.offsetParent;
+    }
+    return offsetParent || document.documentElement;
+  },
 
-  div.style.position = "absolute";
-  div.style.opacity = "0";
-  div.title = "New Privly message";
+  /**
+   * Get the current coordinates of the element, relative to the document.
+   *
+   * jQuery.fn.offset()
+   * Ported from: https://github.com/jquery/jquery/blob/master/src/offset.js
+   * 
+   * @param  {Element} elem
```

Comment:
Include a description of the element, ie, `@param  {Element} elem The element that we are positioning the button relative to.`